sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -2575,6 +2575,22 @@ class AdaptiveQueryExecSuite
       assert(findTopLevelAggregate(adaptive5).size == 4)
     }
   }
+
+  test("SPARK-38578: Avoid unnecessary sort in FileFormatWriter if user has specified sort") {
```

Comment:
We should refine the test name if we don't really test table insertion.

Reply:
yeah, refined it